### ISSUE TITLE
Add option for preferred position side

### ISSUE THREAD

**Issue by MrCasCode** [malikwang/custom_pop_up_menu#13]
Currently preferred position takes only top or bottom as parameter. It would be ideal to support also right / left

**Comment by malikwang** [comment]
like this?
![image](https://user-images.githubusercontent.com/26924433/145137017-b95d74c6-d42e-42b9-9e82-49ae4a072848.png)

**Comment by MrCasCode** [comment]
Exactly! I'm implementing the spectrum components in flutter and the popover on the side is quite common on desktop applications: https://spectrum.adobe.com/page/popover/

**Comment by malikwang** [comment]
> Exactly! I'm implementing the spectrum components in flutter and the popover on the side is quite common on desktop applications: https://spectrum.adobe.com/page/popover/

OK, I understand. I would implement this feature as soon as possible.

**Comment by minhnguyenandpad** [comment]
Do we have any update for this.

**Comment by AhmedAbogameel** [comment]
any updates ?
